deploy/helm/sumologic/templates/logs/otelcol/statefulset.yaml

```diff
@@ -0,0 +1,144 @@
+{{- if eq (include "logs.otelcol.enabled" .) "true" }}
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ template "sumologic.metadata.name.logs.statefulset" . }}
+  labels:
+    app: {{ template "sumologic.labels.app.logs.statefulset" . }}
+    {{- include "sumologic.labels.common" . | nindent 4 }}
+spec:
+  selector:
+    matchLabels:
+      app: {{ template "sumologic.labels.app.logs.pod" . }}
+  serviceName: {{ template "sumologic.metadata.name.logs.service-headless" . }}
+  podManagementPolicy: "Parallel"
+{{- if not .Values.logs.autoscaling.enabled }}
+  replicas: {{ .Values.logs.statefulset.replicaCount }}
+{{- end }}
+  template:
+    metadata:
+      annotations:
+        checksum/config: {{ include (print $.Template.BasePath "/logs/otelcol/configmap.yaml") . | sha256sum }}
+{{- if .Values.sumologic.podAnnotations }}
+{{ toYaml .Values.sumologic.podAnnotations | indent 8 }}
+{{- end }}
+{{- if .Values.otelcol.podAnnotations }}
+{{ toYaml .Values.otelcol.podAnnotations | indent 8 }}
+{{- end }}
+{{- if .Values.logs.statefulset.podAnnotations }}
+{{ toYaml .Values.logs.statefulset.podAnnotations | indent 8 }}
+{{- end }}
+      labels:
+        app: {{ template "sumologic.labels.app.logs.pod" . }}
```

Comment:
This also needs to be changed to take into account the provider. As of now I'm seeing

```
Name:         collection-sumologic-otelcol-logs-0
Namespace:    sumologic
Priority:     0
Node:         sumologic-kubernetes-collection/10.0.2.15
Start Time:   Wed, 08 Sep 2021 12:08:51 +0200
Labels:       app=collection-sumologic-fluentd-logs
              chart=sumologic-2.2.0-dev.0
              controller-revision-hash=collection-sumologic-otelcol-logs-56cf654b8
              heritage=Helm
              release=collection
              statefulset.kubernetes.io/pod-name=collection-sumologic-otelcol-logs-0
...
```